File: fixtures/tiny-go/internal/retry/retry.go
// Package retry runs an operation until it succeeds.
package retry

// DefaultAttempts is how many times Do tries by default.
const DefaultAttempts = 3

// Do runs op until it returns nil or attempts run out.
func Do(attempts int, op func() error) error {
	var err error
	for i := 0; i < attempts; i++ {
		err = op()
		if err == nil {
			return nil
		}
	}
	return err
}

File: fixtures/tiny-ts/packages/core/src/retry.ts
export const DEFAULT_ATTEMPTS = 3;

export interface RetryOptions {
  attempts?: number;
}

export async function retry<T>(fn: () => Promise<T>, opts: RetryOptions = {}): Promise<T> {
  const attempts = opts.attempts ?? DEFAULT_ATTEMPTS;
  let lastError: unknown;
  for (let i = 0; i < attempts; i++) {
    try {
      return await fn();
    } catch (err) {
      lastError = err;
    }
  }
  throw lastError;
}
